Client/State/ClientState.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using BlazorCMS.Shared.Dtos;
+using BlazorState;
+
+namespace BlazorCMS.Client.State
+{
+    public class ClientState : IState
+    {
+        public List<SectionDto> Sections { get; set; }
+        public List<ArticleDto> Articles { get; set; }
+
+        public List<ArticleDto> GetArticlesBySectionId(long sectionId)
+        {
+            return Articles?.Where(e => e.SectionId == sectionId)?.ToList();

```

Comment:
currently this would return null if Articles is null. Would an empty list work better?

Reply:
here I planned to use a null check to determine if the articles are already loaded for the section. so if this returns null, it indicates I need to load the articles via the API.

Reply:
Ah, gotcha! That makes sense, it's one of those things I see and question the implication it has outside of the scope of the function. 

In other words, if I came in and used this, how would I know this function returns null if no articles are found? A comment would suffice to get that across, or maybe it's better to return an empty list instead of null and do an Any check on the result.

Def doesn't matter, and ultimately it's 100% a conventional thing. 👌 Makes total sense!